codenames/guessers/learned_guesser.py

```diff
@@ -0,0 +1,77 @@
+from typing import List
+from overrides import overrides
+
+import torch
+from torch.distributions import Categorical
+
+from codenames.guessers.guesser import Guesser
+from codenames.guessers.policy.guesser_policy import GuesserPolicy
+from codenames.embedding_handler import EmbeddingHandler
+from codenames.utils import game_utils as util
+
+
+class LearnedGuesser(Guesser):
+    def __init__(self,
+                 board: List[str],
+                 embedding_handler: EmbeddingHandler,
+                 policy: GuesserPolicy,
+                 learning_rate: float) -> None:
+        super(LearnedGuesser, self).__init__(board, embedding_handler)
+        self.policy = policy
+        self.guess_history = None
+        self.guess_log_probs = None
+        self.optimizer = torch.optim.Adam(policy.parameters(), lr=learning_rate)
+
+    @overrides
+    def guess(self,
```

Comment:
Looks like we need to reconcile the randomguesser and this one to implement the same abstract class. I'll do that in a separate PR.